common/src/main/java/com/wynntils/features/user/ItemLockFeature.java

```diff
@@ -0,0 +1,160 @@
+        if (!(McUtils.mc().screen instanceof AbstractContainerScreen<?> abstractContainerScreen)) return;
+
+        Character character = WynnUtils.getCharacter();
+        if (character == null) return;
+
+        Character.CharacterInfo characterInfo = character.getCharacterInfo();
+        if (characterInfo == null) return;
+
+        Slot hoveredSlot = abstractContainerScreen.hoveredSlot;
+        if (hoveredSlot == null || !(hoveredSlot.container instanceof Inventory)) return;
+
+        classSlotLockMap.putIfAbsent(characterInfo.getId(), new HashSet<>());
+
+        Set<Integer> classSet = classSlotLockMap.get(characterInfo.getId());
+
+        if (classSet.contains(hoveredSlot.getContainerSlot())) {
+            classSet.remove(hoveredSlot.getContainerSlot());
+        } else {
+            classSet.add(hoveredSlot.getContainerSlot());
+        }
+
+        ConfigManager.saveConfig();
+    }
+
+    @Config(visible = false, internal = true)
+    private static Map<Integer, Set<Integer>> classSlotLockMap = new HashMap<>();
+
+    @Override
+    public void preInit() {
+        addConfigTypeOverride("classSlotLockMap", new TypeToken<HashMap<Integer, Set<Integer>>>() {}.getType());
+    }
+
+    @SubscribeEvent
+    public void onContainerRender(ContainerRenderEvent event) {
+        if (!(event.getScreen() instanceof AbstractContainerScreen<?> abstractContainerScreen)) return;
+
+        Character character = WynnUtils.getCharacter();
+
+        if (character == null || character.getCharacterInfo() == null) return;
+
+        for (Integer slotId :
+                classSlotLockMap.getOrDefault(character.getCharacterInfo().getId(), Set.of())) {
+            Optional<Slot> lockedSlot = abstractContainerScreen.getMenu().slots.stream()
+                    .filter(slot -> slot.container instanceof Inventory && slot.getContainerSlot() == slotId)
+                    .findFirst();
+
+            if (lockedSlot.isEmpty()) {
+                continue;
+            }
+
+            renderLockedSlot(event.getPoseStack(), abstractContainerScreen, lockedSlot.get());
+        }
+    }
+
+    private void renderLockedSlot(PoseStack poseStack, AbstractContainerScreen<?> containerScreen, Slot lockedSlot) {
+        RenderUtils.drawTexturedRect(
+                poseStack,
+                Texture.ITEM_LOCK.resource(),
+                ((containerScreen.leftPos + lockedSlot.x)) + 12,
+                ((containerScreen.topPos + lockedSlot.y)) - 4,
+                400,
+                8,
+                8,
+                Texture.ITEM_LOCK.width() / 2,
+                Texture.ITEM_LOCK.height() / 2);
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onInventoryClickEvent(ContainerClickEvent event) {
+        if (!(McUtils.mc().screen instanceof AbstractContainerScreen<?> abstractContainerScreen)) return;
+        if (event.getClickType() != ClickType.THROW) return;
```

Comment:
i would like to see the full functionality from wynntils legacy ported, i.e. an option to block _all_ interaction with a slot as opposed to just dropping it